testsuite/MDAnalysisTests/transformations/test_translate.py

```diff
@@ -49,6 +49,19 @@ def test_translate_coords(translate_universes):
     trans = translate(vector)(trans_u.trajectory.ts)
     assert_array_almost_equal(trans.positions, ref.positions, decimal=6)
 
+def test_wrap_id(translate_universes):
+    # If the translation is given a different timestep from what is has, the timestep will 
+    # be updated to the new one given. This should only matter in a ChainReader
+    ref_u, trans_u = translate_universes
+    trans_u.dimensions = [363., 364., 365., 90., 90., 90.]
+    ref_u.dimensions = [363., 364., 365., 90., 90., 90.]
+
+    ag = trans_u.residues[24].atoms
+    trans = wrap(ag)
+    id_trans = id(trans_u.trajectory.ts)
+    trans = trans._transform(ref_u.trajectory.ts)
+    assert id_trans != id(trans_u.trajectory.ts) and id(trans_u.trajectory.ts) == id(ref_u.trajectory.ts)
```

Comment:
Just a suggestion, but the first thing I wondered reading over the issue and this PR was why `wrap` might be different from the opposite operation, `unwrap`, so I wrote a short test script that seems to pass on `develop` and fail here (see below). Perhaps a simpler version of that test and a similar change to `unwrap` might be in order? You might also discuss dealing with `unwrap` in a separate PR with the team though, they might be "ok" with that, not sure.

```python
import MDAnalysis
import MDAnalysis.transformations
from MDAnalysisTests.datafiles import PSF, DCD

from numpy.testing import assert_allclose


def main():
    u1 = MDAnalysis.Universe(PSF, DCD, DCD)
    ag1 = u1.atoms
    u2 = MDAnalysis.Universe(PSF, DCD, DCD)

    dim = [2, 2, 2, 90, 90, 90]
    transform_0 = MDAnalysis.transformations.boxdimensions.set_dimensions(dim)
    transform_1 = MDAnalysis.transformations.wrap(ag1)
    transform_2 = MDAnalysis.transformations.unwrap(ag1)
    u1.trajectory.add_transformations(transform_0, transform_1, transform_2)

    for ts1, ts2 in zip(u1.trajectory, u2.trajectory):
        try:
            assert_allclose(ts1.positions, ts2.positions)
        except:
            print("ts1.frame differs:", ts1.frame)



if __name__ == "__main__":
    main()

```

Reply:
That's a good point, it seems like it would be true for any transformation call to _transform that involves an atom group attribute. This would also include fit_rot_trans, while positionaveraging already checks and corrects the timestep.
If I check the other cases for fit_rot_trans and unwrap in the original issue, wouldn't it make more sense to change the issue title to be more broad? Especially since these are so closely related.